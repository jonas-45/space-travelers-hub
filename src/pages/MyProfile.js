import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const { rockets } = useSelector((state) => state.rocket);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  const { missions } = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div>
      <div
        className="m-auto d-flex justify-content-around gap-5"
        style={{ width: '80%' }}
      >
        <div style={{ width: '95%' }}>
          <h4>My Missions</h4>
          <table
            className="table table-bordered"
            style={{ border: '1px solid #dddddd' }}
          >
            {reservedMissions.map((mission) => (
              <tr key={mission.id}>
                <th
                  className="p-2 mx-2"
                  style={{ color: '#444', fontWeight: 'normal' }}
                >
                  {mission.name}
                </th>
              </tr>
            ))}
          </table>
        </div>
        <div style={{ width: '95%' }}>
        <h4>My Rockets</h4>
        {reservedRockets.length <= 0 ? <p>No reserved rockets found!</p> : ''}
        <table className="table table-bordered" style={{ border: '1px solid #dddddd' }}>
          {reservedRockets.map((rocket) => (
            <tr key={rocket.id}><th className="p-2 mx-2" style={{ color: '#444', fontWeight: 'normal' }}>{rocket.name}</th></tr>
          ))}
        </table>
        </div>
      </div>
    </div>
  );
};

export default MyProfile;
