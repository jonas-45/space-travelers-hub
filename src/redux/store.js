import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rocket: rocketReducer,
  },
});

export default store;
