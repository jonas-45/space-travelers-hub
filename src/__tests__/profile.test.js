import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MyProfile from '../pages/MyProfile';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <MyProfile />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
