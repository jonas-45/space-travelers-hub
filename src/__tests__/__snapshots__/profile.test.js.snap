// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="m-auto d-flex justify-content-around gap-5"
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <div
      style={
        Object {
          "width": "95%",
        }
      }
    >
      <h4>
        My Missions
      </h4>
      <table
        className="table table-bordered"
        style={
          Object {
            "border": "1px solid #dddddd",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "width": "95%",
        }
      }
    >
      <h4>
        My Rockets
      </h4>
      <p>
        No reserved rockets found!
      </p>
      <table
        className="table table-bordered"
        style={
          Object {
            "border": "1px solid #dddddd",
          }
        }
      />
    </div>
  </div>
</div>
`;
